text/3579-pki.md

```diff
@@ -0,0 +1,262 @@
+[reference-level-explanation]: #reference-level-explanation
+
+## Signing Console
+[signing-console]: #signing-console
+
+The Rust Infra Team will deploy a new SSH bastion host, `pki-bastion.rust-lang.org`, which shall be used for performing all PKI operations. This system will have available to it a suite of scripts for performing all the common scenarios mentioned in [When the Quorum will be needed][when-the-quorum-will-be-needed]. Current members of the quorum will authenticate to this system utilizing IAM roles via EC2 Instance Connect, which will force them to use their IAM role with two factor authentication (which will double as their role utilized for the actual quorum process).
+
+- This system shall be isolated and powered down except for scheduled quorum events or regular security updates to the system.
+- The Infra team will deploy monitoring for this system to alert on all logins and power events, allowing us central monitoring and control of the quorum bastion.
+- Activity alerts shall be independently sent to all quorum members in the event of accessing or modifying the keys present on the HSM.
+- The Infra team shall limit access to the HSM to only this Signing Console, preventing lateral movement or compromise of the CloudHSM instance from any other system within the Infrastructure.
+
+## `pki.rust-lang.org`
+[pki-rust-lang-org]: #pkirust-langorg
+
+The Rust Infra Team shall stand up a new subdomain for the exposure of transparency logs and operational logs of the PKI. We leave the implementation details of such a system up to the Infra team and project. This system shall store and publicly serve all data which relates to the activities of the Rust CA, its delegate keys and quorum activities.
+
+This publicly accessible data shall be:
+- Written documentation of this RFC and additional documentation on the services, scripts and processes for the quorum model.
+- Access logs to the [Signing Console][signing-console] containing the username and datetime of access. This shall be signed by the Infra key or a delegate of that key.
+- Access logs to the CloudHSM instance, containing IAM role and datetime of access. This shall be signed by the Infra key or a delegate of that key.
+- Written records of all Quorum Events. These shall contain: Event Description, Quorum members present, date/time of beginning and ending the event, cryptographic fingerprints of keys modified, created or revoked. This written record shall be signed independently by each quorum member present for the event.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This will add new complexity for the project to manage, though that new complexity will provide corresponding benefits.
+
+This adds additional questions for future signing and PKI work to be done within the project; in the case of implementing TUF, sigstore, etc. we will have to make additional considerations for how this CA may play into those systems and how it can be included or excluded.
```

Comment:
Right now, we don't know that the RFC will provide corresponding benefits. We have not decided yet how we will sign crates.io and releases, and as the second sentence acknowledges the solution we might end up with might not fit the X.509 CA model.

If the solution we end up with doesn't use an X.509 CA, the benefits of this will be close to zero while adding a lot of overhead.

Reply:
We've added various details in this RFC justifying why we're bridging to the traditional CA model.

That plus posting the subsequent RFC on crates.io index signing should address this.